waves-ext/src/main/scala/com/wavesplatform/transaction/smart/BlockchainContextFixed.scala

```diff
@@ -0,0 +1,63 @@
+package com.wavesplatform.transaction.smart
+
+import cats.Id
+import cats.implicits._
+import com.wavesplatform.common.state.ByteStr
+import com.wavesplatform.lang.directives.DirectiveSet
+import com.wavesplatform.lang.directives.values.{ContentType, ScriptType, StdLibVersion}
+import com.wavesplatform.lang.v1.CTX
+import com.wavesplatform.lang.v1.evaluator.ctx.EvaluationContext
+import com.wavesplatform.lang.v1.evaluator.ctx.impl.waves.WavesContext
+import com.wavesplatform.lang.v1.evaluator.ctx.impl.{CryptoContext, PureContext}
+import com.wavesplatform.lang.v1.traits.Environment
+import com.wavesplatform.lang.{ExecutionError, Global}
+import com.wavesplatform.state._
+import monix.eval.Coeval
+
+import java.util
+
+object BlockchainContextFixed {
+
+  type In = WavesEnvironment.In
+
+  private[this] val cache = new util.HashMap[(StdLibVersion, Boolean, DirectiveSet), CTX[Environment]]()
```

Comment:
Maybe, create a case class for map key? Now it's not obvios what, for example, Boolean means here

Reply:
BlockchainContextFixed & ScriptRunnerFixed were copied from https://github.com/wavesplatform/Waves/pull/3500 . I hope this hack will be removed for node > 1.3.5